components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -2906,28 +2917,27 @@ Int shoc_main_f(Int shcol, Int nlev, Int nlevi, Real dtime, Int nadv, Int npbl,
     isotropy_d    (temp_2d_d[index_counter++]);
 
   view_3d
-    qtracers_d(temp_3d_d[0]);
+    qtracers_f90_d(temp_3d_d[0]);
 
-  // Add temporary slots for solving
-  Kokkos::resize(qtracers_d,
-                 shcol,nlev,ekat::npack<Spack>(num_qtracers+3));
-
-  // shoc_main treats u/v_wind as 1 array.
+  // shoc_main treats u/v_wind as 1 array and
+  // CXX version of shoc qtracers is the transpose of the fortran version..
   const auto nlev_packs = ekat::npack<Spack>(nlev);
   view_3d horiz_wind_d("horiz_wind",shcol,2,nlev_packs);
+  view_3d qtracers_cxx_d("qtracers",shcol,num_qtracers,nlev_packs);
 
-  // Copy wind into single view
   const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(shcol, nlev_packs);
   Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
     const Int i = team.league_rank();
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, nlev), [&] (const Int& k) {
+      const int k_v = k/Spack::n;
+      const int k_p = k%Spack::n;
 
-    const auto u_wind_s = ekat::subview(u_wind_d, i);
-    const auto v_wind_s = ekat::subview(v_wind_d, i);
-    const auto horiz_wind_s = Kokkos::subview(horiz_wind_d,i,Kokkos::ALL(),Kokkos::ALL());
+      horiz_wind_d(i,0,k_v)[k_p] = u_wind_d(i,k_v)[k_p];
```

Comment:
Not a big deal and you don't have to change it here, but I notice that `scalarize` isn't used as often as it could be. `scalarize` leads to simpler code when you need to read and write individual scalars b/c you can omit the div-mod arithmetic.